### PR TITLE
Updates to trackers and global variable handling

### LINKED ISSUE #93: Refactor BIDS tracker
The current [`bids_tracker.py` ](https://github.com/neurodatascience/mr_proc/blob/f4682b400f82c101149281346f4e25a8fc5b0aad/trackers/bids_tracker.py) is very old and needs a big update.

- [ ] Use manifest to know which participants/sessions to track
- [ ] Check BIDS datatype (`anat`, `dwi`, `func`, etc.)
- [ ] Check BIDS suffix (`T1w`, etc.)

@nikhil153 I think we should be able to make a `tracker_configs` dictionary like for the derivatives trackers. It's just that the column names in the bagel will have to be `HAS_DATATYPE__<...>` and `HAS_IMAGE__<...>` instead of `pipeline_complete`/`PHASE__<...>`/`STAGE__<...>` (and the values will be `True`/`False` instead of `SUCCESS`/`FAIL`/`UNAVAILABLE`)

### PR DESCRIPTION
**Pull request by nikhil153**
- [x] updated catalog to use global vars from utils and tracker
- [x] fixed logging levels in mriqc tracker
- [x] refactored bids_tracker - Closes #93 
- [x] updated sample_run_nipoppy to set log-level and prototypical mriqc run with tracker
- [x] updated mriqc and fmirpre run scripts to bind complete bids_dir path
- [x] added sqldb generation with ignore list for subjects and datatype+acq
- [x] updated sample_run_nipoppy.py to showcase use of custom sqldb with mriqc and fmripre

### CODE REVIEW

File: nipoppy/sample_run_nipoppy.py [hunk 1]
```diff
@@ -27,29 +36,85 @@
 log_file = f"{log_dir}/nipoppy.log"
 
 session_id = args.session_id
+session = f"ses-{session_id}"
```
Comment (michellewang) [inline nipoppy/sample_run_nipoppy.py:39]: Can use `nipoppy.workflow.utils.session_id_to_bids_session` here?

File: nipoppy/sample_run_nipoppy.py [hunk 2]
```diff
@@ -27,29 +36,85 @@
...
     elif wf == "bids_conv": 
-        run_bids_conv.run(global_configs, session_id, n_jobs=n_jobs)
+        run_bids_conv.run(global_configs, session_id, n_jobs=n_jobs, logger=logger)
+
+    elif wf == "mriqc":
+        # Supported modalities (i.e. suffixes) for MRIQC
+        modalities = ["T1w", "T2w"]
+        
+        # Run mriqc tracker to regenerate bagel
+        dash_schema_file = f"{DATASET_ROOT}/proc/bagel_schema.json"
```
Comment (michellewang) [inline nipoppy/sample_run_nipoppy.py:79]: Maybe add a comment here saying that the schema needs to be downloaded from the dashboard repo. Or: add the schema to this repo like we said in #97.

**Review by michellewang — APPROVED** [review]
Looks good, just had some minor comments. Ideally all the trackers would take the same inputs for their status-checking function, but maybe we can address that in a future refactor.

I'm a bit concerned with how this PR will integrate with PR #162, which makes kind of big changes (mostly restructuring) to `run_tracker.py`. Might be tricky to merge...

You can also add "Closes #93" in the PR description so that it'll close the issue automatically.

**Comment by nikhil153** [comment]
@michellewang 

> I'm a bit concerned with how this PR will integrate with PR https://github.com/neurodatascience/nipoppy/pull/162, which makes kind of big changes (mostly restructuring) to run_tracker.py. Might be tricky to merge...

My changes to run_tracker.py are minor. If you prefer, I can wait for your PR to be merged.

**Comment by nikhil153** [comment]
@michellewang - the absolute bids_dir path binds work!

**Pull request merged**